### PR TITLE
DPDK telemetry agent feature check in

### PR DESCRIPTION
**Pull request by jianzzha**
The README file contains essential information regarding this check-in, making it a valuable read to understand the mechanism of collecting traffic metrics from a DPDK application.

### CODE REVIEW

File: dpdk-metrics-agent/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# DPDK Telemetry Agent
+
+The DPDK telemetry agent is designed to collect the DPDK traffic metrics where the kernel-based network metrics collecor cannot be applied.
```
Comment (akiselev1) [inline dpdk-metrics-agent/README.md]: collector

File: dpdk-metrics-agent/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## DPDK telemetry API
+
+By default, every DPDK application creates a socket with the path `<runtime_directory>/dpdk_telemetry.<version>`, unless it is disabled during the application's build process. For more details, interested users are encouraged to refer to: https://doc.dpdk.org/guides/howto/telemetry.html.
+
+This agent uses this socket to aquire DPDK traffic metrics from DPDK applications. Through this socket, the agent can monitor a DPDK application's traffic metrics without the need for instrumenting the application.
```
Comment (akiselev1) [inline dpdk-metrics-agent/README.md]: acquire

File: dpdk-metrics-agent/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* --interval, time interval in seconds for metric collection
+
+## Agent deployment in kubernetes
+
+The DPDK telemetry agent can integrate with either Prometheus or OpenTelemetry collector as its backend.
+
+In the diagram below, the agent is deployed as a sidecar in the same pod alongside the DPDK application container, offering a scraping point for Prometheus.
+![Agent deployed sidecar as a prometheus scraping point](../diagrams/dpdk-telemetry-prometheus-scraping.jpg?raw=true "Prometheus Scraping")
+
+In the diagram below, the agent is deployed as a sidecar in the same pod alongside the DPDK application container. It uses OTLP to exports the DPDK traffic metrics to the OpenTelemetry collector.
```
Comment (akiselev1) [inline dpdk-metrics-agent/README.md]: to export

File: dpdk-metrics-agent/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+![Agent deployed sidecar as a prometheus scraping point](../diagrams/dpdk-telemetry-otlp.jpg?raw=true "Prometheus Scraping")
```
Comment (akiselev1) [inline dpdk-metrics-agent/README.md]: This should say Open Telemetry Collector, not Prometheus Scrapping again.

File: diagrams/dpdk-telemetry-prometheus-scraping.jpg
Comment (akiselev1) [inline diagrams/dpdk-telemetry-prometheus-scraping.jpg:1]: What is "service" in this picture?

File: dpdk-metrics-agent/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+## Backend support
+
+* Prometheus
+* OpenTelemetry Collector
+
+## Agent command line options
+
+* --sock-prefix, DPDK telemetry socket directory
+* --port, prometheus scaping port
```
Comment (dkosteck) [inline dpdk-metrics-agent/README.md]: Should be "scraping port"

**Review by dkosteck — CHANGES_REQUESTED** [review]
To me the commits look like great progress, I just had one request for a spelling fix on preliminary review. Perhaps we should have a sync at some point to demo the functionality

File: dpdk-metrics-agent/README.md (hunk 5 above)
Comment (jianzzha) [reply dpdk-metrics-agent/README.md]: Good catch. Have a meeting setup to go over it with demo.

File: dpdk-metrics-agent/README.md (hunk 4 above)
Comment (jianzzha) [reply dpdk-metrics-agent/README.md]: Great catch!

**Review by akiselev1 — APPROVED** [review]

**Review by dkosteck — APPROVED** [review]
Looks good, thanks

**Pull request merged**